javaHapiValidatorLambda/src/main/java/software/amazon/fwoa/Handler.java

```diff
@@ -38,4 +65,50 @@ public ValidatorResponse handleRequest(String event, Context context) {
         return validate;
 
     }
+
+    private List<String> getBucketObjects(AmazonS3 s3, String bucketName, String regionString) {
+
+        ListObjectsV2Result result = s3.listObjectsV2(bucketName);
+        List<S3ObjectSummary> objects = result.getObjectSummaries();
+        List<String> objectKeys = new ArrayList<String>();
+        for (S3ObjectSummary os : objects) {
+            objectKeys.add(os.getKey());
+        }
+        return objectKeys;
+    }
+
+    private String getIGBucketName(List<Bucket> buckets) {
+        for (int i = 0; i < buckets.size(); i += 1) {
+            if (buckets.get(i).getName().contains("fhirimplementationguides")) {
+                return buckets.get(i).getName();
+            }
+        }
+        return "";
+    }
+
+    private String downloadObjects(AmazonS3 s3, String bucketName, List<String> keys) {
+        // create a directory to pass into the validator constructor in /tmp
+        String tmpDir = System.getProperty("java.io.tmpdir");
+        Path tmpDirPath = Paths.get(tmpDir + "/implementationGuides");
+        try {
+            Path igLocation = Files.createDirectory(tmpDirPath);
+            for (int i = 0; i < keys.size(); i += 1) {
+                S3Object guide = s3.getObject(bucketName, keys.get(i));
+                S3ObjectInputStream inputStream = guide.getObjectContent();
+                FileOutputStream outputStream = new FileOutputStream(new File(igLocation.toAbsolutePath().toString() + "/" + keys.get(i)));
+                byte[] buffer = new byte[1024];
+                int length = 0;
+                while ((length = inputStream.read(buffer)) > 0) {
+                    outputStream.write(buffer, 0, length);
+                }
+                inputStream.close();
+                outputStream.close();
+            }
+            return igLocation.toAbsolutePath().toString();
+        } catch (IOException e) {
+            throw new Error(e.getMessage());
+        } catch (AmazonServiceException e) {
+            throw new Error(e.getErrorMessage());
```

Comment:
You don't need to recast the error into a more generic error
```suggestion
        } catch (IOException e) {
            throw e;
        } catch (AmazonServiceException e) {
            throw e;
```